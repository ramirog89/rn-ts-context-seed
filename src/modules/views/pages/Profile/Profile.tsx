import React from 'react';
import { Text, View, Button } from 'react-native';

import { styles } from './styles';

interface IProfileSceenProps {
  navigation: any;
}

const ProfileScreen = ({ navigation }: IProfileSceenProps) => {
  const navigateHome = () => {
    navigation.navigate('Home');
  }
  return (
    <View>
      <View style={styles.sectionContainer}>
        <Text
          style={[
            styles.sectionTitle,
          ]}>
            Aca va mi perfil
            <Button onPress={navigateHome} title="Go back home" />
        </Text>
      </View>
    </View>
  );
}

export default ProfileScreen;
