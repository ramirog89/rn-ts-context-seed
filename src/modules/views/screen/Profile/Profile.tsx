import React from 'react';
import { Text, View } from 'react-native';
import Layout from '../../common/Layout';

import { styles } from './styles';

const ProfileScreen = () => {
  return (
    <Layout>
      <View>
        <View style={styles.sectionContainer}>
          <Text
            style={[
              styles.sectionTitle,
            ]}>
              Aca va mi perfil
          </Text>
        </View>
      </View>
    </Layout>
  );
}

export default ProfileScreen;
