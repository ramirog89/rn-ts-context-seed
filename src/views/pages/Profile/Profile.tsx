import React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';

interface IProfileSceenProps {
  navigation: any;
}

const ProfileScreen = ({ navigation }: IProfileSceenProps) => {
  return (
    <View>
        <View style={styles.sectionContainer}>
            <Text
            style={[
              styles.sectionTitle,
            ]}>
                Aca va mi perfil
                <Button onPress={() => navigation.goBack()} title="Go back home" />
          </Text>
        </View>
    </View>
  );
}

 const styles = StyleSheet.create({
   sectionContainer: {
     marginTop: 32,
     paddingHorizontal: 24,
   },
   sectionTitle: {
     fontSize: 24,
     fontWeight: '600',
   },
 });

export default ProfileScreen;
