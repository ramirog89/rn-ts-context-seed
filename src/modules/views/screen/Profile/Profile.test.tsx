import React from 'react';

import { mockedRoute } from '../../../../test/setup';
import { render } from '../../../../test/render';

import Profile from './Profile';

describe('ProfileScreen', () => {
  beforeEach(() => {
    mockedRoute.mockReturnValue('Profile');
  });

  it('should render', () => {
    const { toJSON } = render(<Profile />);
    expect(toJSON()).toMatchSnapshot();
  });
});
