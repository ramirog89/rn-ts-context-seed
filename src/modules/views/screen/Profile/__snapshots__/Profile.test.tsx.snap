// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "#29434e",
        }
      }
    >
      <View
        noLeft={false}
        style={
          Array [
            Object {
              "backgroundColor": "#F8F8F8",
              "borderBottomColor": "#a7a6ab",
              "borderBottomWidth": 0.5,
              "elevation": 3,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 6,
              "paddingRight": 10,
              "right": 0,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "top": 0,
            },
            Object {
              "backgroundColor": "#29434e",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginLeft": -3,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 4,
                "paddingRight": 6,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
            testID="goBackButton"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 34,
                      "marginLeft": 1,
                      "marginRight": 2,
                      "marginTop": 0,
                      "paddingTop": 1,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "700",
                "marginLeft": undefined,
                "paddingLeft": 4,
                "paddingTop": 1,
                "textAlign": "center",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 50,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginRight": -9,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
            testID="openMenuButton"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 31,
                      "marginLeft": 2,
                      "marginRight": 0,
                      "marginTop": 0,
                      "paddingTop": 2,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          undefined,
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          },
          Object {
            "padding": 5,
          },
        ]
      }
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View>
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              Profile Screen
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
